diff -up linux-6.13-rc1/drivers/mfd/rk8xx-core.c.94~ linux-6.13-rc1/drivers/mfd/rk8xx-core.c
--- linux-6.13-rc1/drivers/mfd/rk8xx-core.c.94~	2024-12-02 01:09:26.043799984 +0100
+++ linux-6.13-rc1/drivers/mfd/rk8xx-core.c	2024-12-02 01:10:02.644233878 +0100
@@ -788,7 +788,7 @@ int rk8xx_probe(struct device *dev, int
 	if (device_property_read_bool(dev, "system-power-controller") ||
 	    device_property_read_bool(dev, "rockchip,system-power-controller")) {
 		ret = devm_register_sys_off_handler(dev,
-				    SYS_OFF_MODE_POWER_OFF_PREPARE, SYS_OFF_PRIO_HIGH,
+				    SYS_OFF_MODE_POWER_OFF, SYS_OFF_PRIO_HIGH,
 				    &rk808_power_off, rk808);
 		if (ret)
 			return dev_err_probe(dev, ret,
