diff -up linux-6.13-rc1/arch/arm64/boot/dts/rockchip/rk3588s-coolpi-4b.dts.104~ linux-6.13-rc1/arch/arm64/boot/dts/rockchip/rk3588s-coolpi-4b.dts
--- linux-6.13-rc1/arch/arm64/boot/dts/rockchip/rk3588s-coolpi-4b.dts.104~	2024-12-02 01:13:31.216706143 +0100
+++ linux-6.13-rc1/arch/arm64/boot/dts/rockchip/rk3588s-coolpi-4b.dts	2024-12-02 01:15:00.357762617 +0100
@@ -184,6 +184,18 @@
 		vin-supply = <&vcc5v0_sys>;
 	};
 
+	vcc1v8_wifi: vcc1v8-wifi-regulator {
+		compatible = "regulator-fixed";
+		enable-active-high;
+		gpio = <&gpio1 RK_PD3 GPIO_ACTIVE_HIGH>;
+		regulator-name = "vcc3v3_wifi";
+		regulator-boot-on;
+		regulator-always-on;
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		vin-supply = <&vcc5v0_sys>;
+	};
+
 	vcc_1v1_nldo_s3: regulator-vcc-1v1-nldo-s3 {
 		compatible = "regulator-fixed";
 		regulator-name = "vcc_1v1_nldo_s3";
