diff -up linux-6.4-rc1/drivers/pci/pci.c.56~ linux-6.4-rc1/drivers/pci/pci.c
--- linux-6.4-rc1/drivers/pci/pci.c.56~	2023-05-08 20:45:52.594138998 +0200
+++ linux-6.4-rc1/drivers/pci/pci.c	2023-05-08 20:47:38.274958457 +0200
@@ -62,7 +62,7 @@ struct pci_pme_device {
 	struct pci_dev *dev;
 };
 
-#define PME_TIMEOUT 1000 /* How long between PME checks */
+#define PME_TIMEOUT 4000 /* How long between PME checks */
 
 /*
  * Devices may extend the 1 sec period through Request Retry Status
