diff -up linux-6.5-rc1/drivers/pci/pci.c.56~ linux-6.5-rc1/drivers/pci/pci.c
--- linux-6.5-rc1/drivers/pci/pci.c.56~	2023-07-11 11:51:51.764578834 +0200
+++ linux-6.5-rc1/drivers/pci/pci.c	2023-07-11 11:52:43.424099055 +0200
@@ -62,7 +62,7 @@ struct pci_pme_device {
 	struct pci_dev *dev;
 };
 
-#define PME_TIMEOUT 1000 /* How long between PME checks */
+#define PME_TIMEOUT 4000 /* How long between PME checks */
 
 /*
  * Following exit from Conventional Reset, devices must be ready within 1 sec
