diff -up linux-6.0-rc1/net/ipv4/tcp.c.48~ linux-6.0-rc1/net/ipv4/tcp.c
--- linux-6.0-rc1/net/ipv4/tcp.c.48~	2022-08-15 14:16:44.494026534 +0200
+++ linux-6.0-rc1/net/ipv4/tcp.c	2022-08-15 14:18:22.288260450 +0200
@@ -4783,8 +4783,8 @@ void __init tcp_init(void)
 	tcp_init_mem();
 	/* Set per-socket limits to no more than 1/128 the pressure threshold */
 	limit = nr_free_buffer_pages() << (PAGE_SHIFT - 7);
-	max_wshare = min(4UL*1024*1024, limit);
-	max_rshare = min(6UL*1024*1024, limit);
+	max_wshare = min(16UL*1024*1024, limit);
+	max_rshare = min(16UL*1024*1024, limit);
 
 	init_net.ipv4.sysctl_tcp_wmem[0] = PAGE_SIZE;
 	init_net.ipv4.sysctl_tcp_wmem[1] = 16*1024;
