diff -up linux-6.16-rc2/net/ipv4/tcp.c.57~ linux-6.16-rc2/net/ipv4/tcp.c
--- linux-6.16-rc2/net/ipv4/tcp.c.57~	2025-06-15 22:49:41.000000000 +0200
+++ linux-6.16-rc2/net/ipv4/tcp.c	2025-06-18 16:06:40.726133163 +0200
@@ -5226,7 +5226,7 @@ void __init tcp_init(void)
 	tcp_init_mem();
 	/* Set per-socket limits to no more than 1/128 the pressure threshold */
 	limit = nr_free_buffer_pages() << (PAGE_SHIFT - 7);
-	max_wshare = min(4UL*1024*1024, limit);
+	max_wshare = min(16UL*1024*1024, limit);
 	max_rshare = min(32UL*1024*1024, limit);
 
 	init_net.ipv4.sysctl_tcp_wmem[0] = PAGE_SIZE;
