diff -up linux-7.1-rc3/drivers/net/dummy.c.60~ linux-7.1-rc3/drivers/net/dummy.c
--- linux-7.1-rc3/drivers/net/dummy.c.60~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/drivers/net/dummy.c	2026-05-17 22:28:08.471328600 +0200
@@ -44,7 +44,7 @@
 
 #define DRV_NAME	"dummy"
 
-static int numdummies = 1;
+static int numdummies = 0;
 
 /* fake multicast ability */
 static void set_multicast_list(struct net_device *dev,
diff -up linux-7.1-rc3/include/uapi/linux/if_bonding.h.60~ linux-7.1-rc3/include/uapi/linux/if_bonding.h
--- linux-7.1-rc3/include/uapi/linux/if_bonding.h.60~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/include/uapi/linux/if_bonding.h	2026-05-17 22:20:48.621598903 +0200
@@ -82,7 +82,7 @@
 #define BOND_STATE_ACTIVE       0   /* link is active */
 #define BOND_STATE_BACKUP       1   /* link is backup */
 
-#define BOND_DEFAULT_MAX_BONDS  1   /* Default maximum number of devices to support */
+#define BOND_DEFAULT_MAX_BONDS  0   /* Default maximum number of devices to support */
 
 #define BOND_DEFAULT_TX_QUEUES 16   /* Default number of tx queues per device */
 
