diff -up linux-5.10-rc7/mm/filemap.c.omv~ linux-5.10-rc7/mm/filemap.c
--- linux-5.10-rc7/mm/filemap.c.omv~	2020-12-08 21:02:54.165257854 +0100
+++ linux-5.10-rc7/mm/filemap.c	2020-12-08 21:05:08.112260452 +0100
@@ -827,7 +827,7 @@ int replace_page_cache_page(struct page
 }
 EXPORT_SYMBOL_GPL(replace_page_cache_page);
 
-static noinline int __add_to_page_cache_locked(struct page *page,
+noinline int __add_to_page_cache_locked(struct page *page,
 					struct address_space *mapping,
 					pgoff_t offset, gfp_t gfp,
 					void **shadowp)
