diff -up linux-5.6-rc2/drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c.omv~ linux-5.6-rc2/drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c
--- linux-5.6-rc2/drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c.omv~	2020-02-22 03:48:01.477787558 +0100
+++ linux-5.6-rc2/drivers/gpu/drm/i915/gem/i915_gem_execbuffer.c	2020-02-22 03:48:25.628786386 +0100
@@ -1426,7 +1426,7 @@ static int eb_relocate_vma(struct i915_e
 	struct drm_i915_gem_relocation_entry stack[N_RELOC(512)];
 	struct drm_i915_gem_relocation_entry __user *urelocs;
 	const struct drm_i915_gem_exec_object2 *entry = exec_entry(eb, vma);
-	unsigned int remain;
+	unsigned long remain;
 
 	urelocs = u64_to_user_ptr(entry->relocs_ptr);
 	remain = entry->relocation_count;
