diff -up linux-7.0-rc1/kernel/signal.c.omv~ linux-7.0-rc1/kernel/signal.c
--- linux-7.0-rc1/kernel/signal.c.omv~	2026-02-24 19:31:01.754478945 +0100
+++ linux-7.0-rc1/kernel/signal.c	2026-02-24 19:31:18.136613520 +0100
@@ -1386,7 +1386,7 @@ struct sighand_struct *lock_task_sighand
 
 	return sighand;
 }
-EXPORT_SYMBOL_GPL(__lock_task_sighand);
+EXPORT_SYMBOL_GPL(lock_task_sighand);
 
 #ifdef CONFIG_LOCKDEP
 void lockdep_assert_task_sighand_held(struct task_struct *task)
