diff -up linux-7.0-rc1/drivers/gpu/drm/evdi/evdi_fb.c.omv~ linux-7.0-rc1/drivers/gpu/drm/evdi/evdi_fb.c
--- linux-7.0-rc1/drivers/gpu/drm/evdi/evdi_fb.c.omv~	2026-02-24 23:39:51.894034446 +0100
+++ linux-7.0-rc1/drivers/gpu/drm/evdi/evdi_fb.c	2026-02-24 23:39:57.314800410 +0100
@@ -213,8 +213,8 @@ static const struct fb_ops evdifb_ops =
 	.fb_pan_display = drm_fb_helper_pan_display,
 	.fb_blank = drm_fb_helper_blank,
 	.fb_setcmap = drm_fb_helper_setcmap,
-	.fb_debug_enter = drm_fb_helper_debug_enter,
-	.fb_debug_leave = drm_fb_helper_debug_leave,
+//	.fb_debug_enter = drm_fb_helper_debug_enter,
+//	.fb_debug_leave = drm_fb_helper_debug_leave,
 	.fb_mmap = evdi_fb_mmap,
 	.fb_open = evdi_fb_open,
 	.fb_release = evdi_fb_release,
diff -up linux-7.0-rc1/drivers/gpu/drm/vboxvideo/vbox_fb.c.omv~ linux-7.0-rc1/drivers/gpu/drm/vboxvideo/vbox_fb.c
--- linux-7.0-rc1/drivers/gpu/drm/vboxvideo/vbox_fb.c.omv~	2026-02-24 23:34:04.935691652 +0100
+++ linux-7.0-rc1/drivers/gpu/drm/vboxvideo/vbox_fb.c	2026-02-24 23:34:11.869764207 +0100
@@ -211,8 +211,8 @@ static struct fb_ops vboxfb_ops = {
 	.fb_pan_display = drm_fb_helper_pan_display,
 	.fb_blank = drm_fb_helper_blank,
 	.fb_setcmap = drm_fb_helper_setcmap,
-	.fb_debug_enter = drm_fb_helper_debug_enter,
-	.fb_debug_leave = drm_fb_helper_debug_leave,
+//	.fb_debug_enter = drm_fb_helper_debug_enter,
+//	.fb_debug_leave = drm_fb_helper_debug_leave,
 };
 
 static int vboxfb_create_object(struct vbox_fbdev *fbdev,
diff -up linux-7.0-rc1/kernel/signal.c.omv~ linux-7.0-rc1/kernel/signal.c
--- linux-7.0-rc1/kernel/signal.c.omv~	2026-02-24 19:31:01.754478945 +0100
+++ linux-7.0-rc1/kernel/signal.c	2026-02-24 19:31:18.136613520 +0100
@@ -1386,7 +1386,7 @@ struct sighand_struct *lock_task_sighand
 
 	return sighand;
 }
-EXPORT_SYMBOL_GPL(__lock_task_sighand);
+EXPORT_SYMBOL_GPL(lock_task_sighand);
 
 #ifdef CONFIG_LOCKDEP
 void lockdep_assert_task_sighand_held(struct task_struct *task)
