diff -up linux-6.9-rc1/drivers/platform/x86/acer-wmi.c.omv~ linux-6.9-rc1/drivers/platform/x86/acer-wmi.c
--- linux-6.9-rc1/drivers/platform/x86/acer-wmi.c.omv~	2024-03-27 15:16:33.458312706 +0100
+++ linux-6.9-rc1/drivers/platform/x86/acer-wmi.c	2024-03-27 15:18:28.104893501 +0100
@@ -622,6 +622,15 @@ static const struct dmi_system_id acer_q
 		.driver_data = &quirk_acer_predator_v4,
 	},
 	{
+		.callback = dmi_matched,
+		.ident = "Acer Predator PH18-71",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Acer"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Predator PH18-71"),
+		},
+		.driver_data = &quirk_acer_predator_v4,
+	},
+	{
 		.callback = set_force_caps,
 		.ident = "Acer Aspire Switch 10E SW3-016",
 		.matches = {
