--- a/drivers/gpu/drm/amd/amdgpu/amdgpu.h	2026-08-31 00:33:08.337033106 +0200
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu.h	2026-08-31 00:33:08.363971569 +0200
@@ -157,6 +157,7 @@ struct amdgpu_watchdog_timer {
  */
 extern int amdgpu_modeset;
 extern unsigned int amdgpu_vram_limit;
+extern int amdgpu_ignore_min_pcap;
 extern int amdgpu_vis_vram_limit;
 extern int amdgpu_gart_size;
 extern int amdgpu_gtt_size;
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c	2026-08-31 00:33:08.340158156 +0200
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c	2026-08-31 00:33:08.364305333 +0200
@@ -150,6 +150,7 @@ enum AMDGPU_DEBUG_MASK {
 };
 
 unsigned int amdgpu_vram_limit = UINT_MAX;
+int amdgpu_ignore_min_pcap = 0; /* do not ignore by default */
 int amdgpu_vis_vram_limit;
 int amdgpu_gart_size = -1; /* auto */
 int amdgpu_gtt_size = -1; /* auto */
@@ -272,6 +273,15 @@ struct amdgpu_watchdog_timer amdgpu_watc
 };
 
 /**
+ * DOC: ignore_min_pcap (int)
+ * Ignore the minimum power cap.
+ * Useful on graphics cards where the minimum power cap is very high.
+ * The default is 0 (Do not ignore).
+ */
+MODULE_PARM_DESC(ignore_min_pcap, "Ignore the minimum power cap");
+module_param_named(ignore_min_pcap, amdgpu_ignore_min_pcap, int, 0600);
+
+/**
  * DOC: vramlimit (int)
  * Restrict the total amount of VRAM in MiB for testing.  The default is 0 (Use full VRAM).
  */
--- a/drivers/gpu/drm/amd/pm/amdgpu_pm.c	2026-08-31 00:33:08.343069813 +0200
+++ b/drivers/gpu/drm/amd/pm/amdgpu_pm.c	2026-08-31 00:33:08.364646558 +0200
@@ -3422,6 +3422,9 @@ static ssize_t amdgpu_hwmon_show_power_c
 					 struct device_attribute *attr,
 					 char *buf)
 {
+	if (amdgpu_ignore_min_pcap)
+		return sysfs_emit(buf, "%i\n", 0);
+
 	return amdgpu_hwmon_show_power_cap_generic(dev, attr, buf, PP_PWR_LIMIT_MIN);
 }
 
--- a/drivers/gpu/drm/amd/pm/swsmu/amdgpu_smu.c	2026-08-31 00:33:08.345849128 +0200
+++ b/drivers/gpu/drm/amd/pm/swsmu/amdgpu_smu.c	2026-08-31 00:33:08.364835540 +0200
@@ -3001,9 +3001,12 @@ int smu_get_ppt_limit(void *handle,
 			smu->ppt_limits.range[power_source][limit_type].max;
 		break;
 	case SMU_PPT_LIMIT_MIN:
-		*limit = smu->od_enabled ?
-			smu->ppt_limits.range[power_source][limit_type].od_min :
-			smu->ppt_limits.range[power_source][limit_type].min;
+		if (amdgpu_ignore_min_pcap)
+			*limit = 0;
+		else
+			*limit = smu->od_enabled ?
+				smu->ppt_limits.range[power_source][limit_type].od_min :
+				smu->ppt_limits.range[power_source][limit_type].min;
 		break;
 	default:
 		return -EINVAL;
@@ -3040,7 +3043,7 @@ static int smu_set_ppt_limit(void *handl
 			return ret;
 	}
 
-	if (limit > max_limit || limit < min_limit) {
+	if (limit > max_limit || (!amdgpu_ignore_min_pcap && limit < min_limit)) {
 		dev_err(smu->adev->dev,
 			"New PPT limit (%d) is out of range [%d,%d]\n",
 			limit, min_limit, max_limit);
