diff -up linux-6.19-rc7/arch/x86/kernel/cpu/amd.c.omv~ linux-6.19-rc7/arch/x86/kernel/cpu/amd.c
--- linux-6.19-rc7/arch/x86/kernel/cpu/amd.c.omv~	2026-01-30 02:05:08.858869357 +0100
+++ linux-6.19-rc7/arch/x86/kernel/cpu/amd.c	2026-01-30 02:06:11.445042277 +0100
@@ -1051,9 +1051,9 @@ static const struct x86_cpu_id zen5_rdse
 static void init_amd_zen5(struct cpuinfo_x86 *c)
 {
 	if (!x86_match_min_microcode_rev(zen5_rdseed_microcode)) {
-		clear_cpu_cap(c, X86_FEATURE_RDSEED);
-		msr_clear_bit(MSR_AMD64_CPUID_FN_7, 18);
-		pr_emerg_once("RDSEED32 is broken. Disabling the corresponding CPUID bit.\n");
+/*		clear_cpu_cap(c, X86_FEATURE_RDSEED);
+		msr_clear_bit(MSR_AMD64_CPUID_FN_7, 18); */
+		pr_emerg_once("RDSEED32 is broken. This may result in unsafe crypto. Update your BIOS ASAP!\n");
 	}
 }
 
