diff -up linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_drm_drv.c.omv~ linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_drm_drv.c
--- linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_drm_drv.c.omv~	2023-09-25 20:22:22.567550029 +0200
+++ linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_drm_drv.c	2023-09-25 20:28:32.408099043 +0200
@@ -124,9 +124,11 @@ static struct drm_driver driver = {
 
 	.fops = &evdi_driver_fops,
 
+#if KERNEL_VERSION(6, 6, 0) > LINUX_VERSION_CODE
 	.prime_fd_to_handle = drm_gem_prime_fd_to_handle,
-	.gem_prime_import = drm_gem_prime_import,
 	.prime_handle_to_fd = drm_gem_prime_handle_to_fd,
+#endif
+	.gem_prime_import = drm_gem_prime_import,
 #if KERNEL_VERSION(5, 11, 0) <= LINUX_VERSION_CODE || defined(EL8)
 #else
 	.preclose = evdi_driver_preclose,
diff -up linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_fb.c.omv~ linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_fb.c
--- linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_fb.c.omv~	2023-09-25 20:29:53.644299637 +0200
+++ linux-6.6-rc3/drivers/gpu/drm/evdi/evdi_fb.c	2023-09-25 20:35:56.422352860 +0200
@@ -417,11 +417,13 @@ static int evdifb_create(struct drm_fb_h
 	info->fix.smem_len = size;
 	info->fix.smem_start = (unsigned long)efbdev->efb.obj->vmapping;
 
+#if KERNEL_VERSION(6, 6, 0) > LINUX_VERSION_CODE
 #if KERNEL_VERSION(4, 20, 0) <= LINUX_VERSION_CODE || defined(EL8)
 	info->flags = FBINFO_DEFAULT;
 #else
 	info->flags = FBINFO_DEFAULT | FBINFO_CAN_FORCE_OUTPUT;
 #endif
+#endif
 
 	efbdev->fb_ops = &evdifb_ops;
 	info->fbops = efbdev->fb_ops;
