diff -up linux-6.14-rc2/drivers/gpu/drm/evdi/evdi_drm_drv.c.omv~ linux-6.14-rc2/drivers/gpu/drm/evdi/evdi_drm_drv.c
--- linux-6.14-rc2/drivers/gpu/drm/evdi/evdi_drm_drv.c.omv~	2025-02-15 15:28:14.750935583 +0100
+++ linux-6.14-rc2/drivers/gpu/drm/evdi/evdi_drm_drv.c	2025-02-15 15:28:36.992947952 +0100
@@ -147,7 +147,9 @@ static struct drm_driver driver = {
 
 	.name = DRIVER_NAME,
 	.desc = DRIVER_DESC,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6,14,0)
 	.date = DRIVER_DATE,
+#endif
 	.major = DRIVER_MAJOR,
 	.minor = DRIVER_MINOR,
 	.patchlevel = DRIVER_PATCH,
