diff -up linux-7.1-rc3/fs/file.c.31~ linux-7.1-rc3/fs/file.c
--- linux-7.1-rc3/fs/file.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/fs/file.c	2026-05-17 22:18:06.970423548 +0200
@@ -887,6 +887,7 @@ struct file *file_close_fd(unsigned int
 
 	return file;
 }
+EXPORT_SYMBOL_GPL(file_close_fd);
 
 void do_close_on_exec(struct files_struct *files)
 {
diff -up linux-7.1-rc3/ipc/msgutil.c.31~ linux-7.1-rc3/ipc/msgutil.c
--- linux-7.1-rc3/ipc/msgutil.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/ipc/msgutil.c	2026-05-17 22:18:06.970684701 +0200
@@ -30,6 +30,7 @@ struct ipc_namespace init_ipc_ns = {
 	.ns = NS_COMMON_INIT(init_ipc_ns),
 	.user_ns = &init_user_ns,
 };
+EXPORT_SYMBOL_GPL(init_ipc_ns);
 
 struct msg_msgseg {
 	struct msg_msgseg *next;
diff -up linux-7.1-rc3/ipc/namespace.c.31~ linux-7.1-rc3/ipc/namespace.c
--- linux-7.1-rc3/ipc/namespace.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/ipc/namespace.c	2026-05-17 22:18:06.970822193 +0200
@@ -210,6 +210,7 @@ void put_ipc_ns(struct ipc_namespace *ns
 			schedule_work(&free_ipc_work);
 	}
 }
+EXPORT_SYMBOL_GPL(put_ipc_ns);
 
 static struct ns_common *ipcns_get(struct task_struct *task)
 {
diff -up linux-7.1-rc3/kernel/sched/syscalls.c.31~ linux-7.1-rc3/kernel/sched/syscalls.c
--- linux-7.1-rc3/kernel/sched/syscalls.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/kernel/sched/syscalls.c	2026-05-17 22:18:06.971022505 +0200
@@ -119,6 +119,7 @@ int can_nice(const struct task_struct *p
 {
 	return is_nice_reduction(p, nice) || capable(CAP_SYS_NICE);
 }
+EXPORT_SYMBOL_GPL(can_nice);
 
 #ifdef __ARCH_WANT_SYS_NICE
 
diff -up linux-7.1-rc3/kernel/sched/wait.c.31~ linux-7.1-rc3/kernel/sched/wait.c
--- linux-7.1-rc3/kernel/sched/wait.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/kernel/sched/wait.c	2026-05-17 22:18:06.971254788 +0200
@@ -231,6 +231,7 @@ void __wake_up_pollfree(struct wait_queu
 	/* POLLFREE must have cleared the queue. */
 	WARN_ON_ONCE(waitqueue_active(wq_head));
 }
+EXPORT_SYMBOL_GPL(__wake_up_pollfree);
 
 /*
  * Note: we use "set_current_state()" _after_ the wait-queue add,
diff -up linux-7.1-rc3/kernel/signal.c.31~ linux-7.1-rc3/kernel/signal.c
--- linux-7.1-rc3/kernel/signal.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/kernel/signal.c	2026-05-17 22:18:06.971663072 +0200
@@ -1384,6 +1384,7 @@ struct sighand_struct *lock_task_sighand
 
 	return sighand;
 }
+EXPORT_SYMBOL_GPL(__lock_task_sighand);
 
 #ifdef CONFIG_LOCKDEP
 void lockdep_assert_task_sighand_held(struct task_struct *task)
diff -up linux-7.1-rc3/kernel/task_work.c.31~ linux-7.1-rc3/kernel/task_work.c
--- linux-7.1-rc3/kernel/task_work.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/kernel/task_work.c	2026-05-17 22:18:06.972039986 +0200
@@ -102,6 +102,7 @@ int task_work_add(struct task_struct *ta
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(task_work_add);
 
 /**
  * task_work_cancel_match - cancel a pending work added by task_work_add()
diff -up linux-7.1-rc3/mm/memory.c.31~ linux-7.1-rc3/mm/memory.c
diff -up linux-7.1-rc3/mm/shmem.c.31~ linux-7.1-rc3/mm/shmem.c
--- linux-7.1-rc3/mm/shmem.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/mm/shmem.c	2026-05-17 22:18:06.973739695 +0200
@@ -5959,6 +5959,7 @@ int shmem_zero_setup_desc(struct vm_area
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(shmem_zero_setup);
 
 /**
  * shmem_read_folio_gfp - read into page cache, using specified page allocation flags.
diff -up linux-7.1-rc3/mm/vmalloc.c.31~ linux-7.1-rc3/mm/vmalloc.c
--- linux-7.1-rc3/mm/vmalloc.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/mm/vmalloc.c	2026-05-17 22:18:06.974303871 +0200
@@ -3270,6 +3270,7 @@ struct vm_struct *get_vm_area(unsigned l
 				  NUMA_NO_NODE, GFP_KERNEL,
 				  __builtin_return_address(0));
 }
+EXPORT_SYMBOL_GPL(get_vm_area);
 
 struct vm_struct *get_vm_area_caller(unsigned long size, unsigned long flags,
 				const void *caller)
diff -up linux-7.1-rc3/security/security.c.31~ linux-7.1-rc3/security/security.c
--- linux-7.1-rc3/security/security.c.31~	2026-05-10 23:08:09.000000000 +0200
+++ linux-7.1-rc3/security/security.c	2026-05-17 22:18:06.974843917 +0200
@@ -514,6 +514,7 @@ int security_binder_set_context_mgr(cons
 {
 	return call_int_hook(binder_set_context_mgr, mgr);
 }
+EXPORT_SYMBOL_GPL(security_binder_set_context_mgr);
 
 /**
  * security_binder_transaction() - Check if a binder transaction is allowed
@@ -529,6 +530,7 @@ int security_binder_transaction(const st
 {
 	return call_int_hook(binder_transaction, from, to);
 }
+EXPORT_SYMBOL_GPL(security_binder_transaction);
 
 /**
  * security_binder_transfer_binder() - Check if a binder transfer is allowed
@@ -544,6 +546,7 @@ int security_binder_transfer_binder(cons
 {
 	return call_int_hook(binder_transfer_binder, from, to);
 }
+EXPORT_SYMBOL_GPL(security_binder_transfer_binder);
 
 /**
  * security_binder_transfer_file() - Check if a binder file xfer is allowed
@@ -560,6 +563,7 @@ int security_binder_transfer_file(const
 {
 	return call_int_hook(binder_transfer_file, from, to, file);
 }
+EXPORT_SYMBOL_GPL(security_binder_transfer_file);
 
 /**
  * security_ptrace_access_check() - Check if tracing is allowed
