diff -up linux-6.3-rc2/fs/f2fs/data.c.32~ linux-6.3-rc2/fs/f2fs/data.c
--- linux-6.3-rc2/fs/f2fs/data.c.32~	2023-03-14 03:15:59.849529893 +0100
+++ linux-6.3-rc2/fs/f2fs/data.c	2023-03-14 03:18:19.298730014 +0100
@@ -1329,18 +1329,14 @@ struct page *f2fs_get_lock_data_page(str
 {
 	struct address_space *mapping = inode->i_mapping;
 	struct page *page;
-repeat:
+
 	page = f2fs_get_read_data_page(inode, index, 0, for_write, NULL);
 	if (IS_ERR(page))
 		return page;
 
 	/* wait for read completion */
 	lock_page(page);
-	if (unlikely(page->mapping != mapping)) {
-		f2fs_put_page(page, 1);
-		goto repeat;
-	}
-	if (unlikely(!PageUptodate(page))) {
+	if (unlikely(page->mapping != mapping || !PageUptodate(page))) {
 		f2fs_put_page(page, 1);
 		return ERR_PTR(-EIO);
 	}
