diff -Naur a/fs/f2fs/data.c b/fs/f2fs/data.c
--- a/fs/f2fs/data.c	2022-12-11 22:15:18.000000000 +0000
+++ b/fs/f2fs/data.c	2023-02-14 18:58:46.292402673 +0000
@@ -1316,18 +1316,14 @@
 {
 	struct address_space *mapping = inode->i_mapping;
 	struct page *page;
-repeat:
+
 	page = f2fs_get_read_data_page(inode, index, 0, for_write);
 	if (IS_ERR(page))
 		return page;
 
 	/* wait for read completion */
 	lock_page(page);
-	if (unlikely(page->mapping != mapping)) {
-		f2fs_put_page(page, 1);
-		goto repeat;
-	}
-	if (unlikely(!PageUptodate(page))) {
+	if (unlikely(page->mapping != mapping || !PageUptodate(page))) {
 		f2fs_put_page(page, 1);
 		return ERR_PTR(-EIO);
 	}
