diff -up linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/fw/dbg.c.21~ linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/fw/dbg.c
--- linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/fw/dbg.c.21~	2020-09-25 18:29:47.472795473 +0200
+++ linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/fw/dbg.c	2020-09-25 18:30:30.169441163 +0200
@@ -2554,7 +2554,7 @@ int iwl_fw_start_dbg_conf(struct iwl_fw_
 		return -EINVAL;
 
 	if (fwrt->dump.conf != FW_DBG_INVALID)
-		IWL_INFO(fwrt, "FW already configured (%d) - re-configuring\n",
+		IWL_DEBUG_FW_INFO(fwrt, "FW already configured (%d) - re-configuring\n",
 			 fwrt->dump.conf);
 
 	/* Send all HCMDs for configuring the FW debug */
diff -up linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/pcie/trans.c.21~ linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/pcie/trans.c
--- linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/pcie/trans.c.21~	2020-09-21 01:33:55.000000000 +0200
+++ linux-5.9-rc6/drivers/net/wireless/intel/iwlwifi/pcie/trans.c	2020-09-25 18:29:47.472795473 +0200
@@ -929,7 +929,7 @@ void iwl_pcie_apply_destination(struct i
 		return;
 	}
 
-	IWL_INFO(trans, "Applying debug destination %s\n",
+	IWL_DEBUG_INFO(trans, "Applying debug destination %s\n",
 		 get_fw_dbg_mode_string(dest->monitor_mode));
 
 	if (dest->monitor_mode == EXTERNAL_MODE)
