diff -up linux-6.10-rc1/drivers/gpu/drm/arm/display/komeda/komeda_dev.c.omv~ linux-6.10-rc1/drivers/gpu/drm/arm/display/komeda/komeda_dev.c
--- linux-6.10-rc1/drivers/gpu/drm/arm/display/komeda/komeda_dev.c.omv~	2024-05-28 23:08:27.829791374 +0100
+++ linux-6.10-rc1/drivers/gpu/drm/arm/display/komeda/komeda_dev.c	2024-05-28 23:08:31.869800570 +0100
@@ -14,8 +14,8 @@
 #include <linux/dma-mapping.h>
 #ifdef CONFIG_DEBUG_FS
 #include <linux/debugfs.h>
-#include <linux/seq_file.h>
 #endif
+#include <linux/seq_file.h>
 
 #include <drm/drm_print.h>
 
