diff -up linux-6.9-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c.28~ linux-6.9-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
--- linux-6.9-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c.28~	2024-03-25 12:27:48.258221857 +0100
+++ linux-6.9-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c	2024-03-25 12:29:26.903747981 +0100
@@ -589,18 +589,12 @@ module_param_named(timeout_period, amdgp
 /**
  * DOC: si_support (int)
  * Set SI support driver. This parameter works after set config CONFIG_DRM_AMDGPU_SI. For SI asic, when radeon driver is enabled,
- * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using radeon driver when it available,
- * otherwise using amdgpu driver.
+ * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using amdgpu driver.
  */
 #ifdef CONFIG_DRM_AMDGPU_SI
 
-#if IS_ENABLED(CONFIG_DRM_RADEON) || IS_ENABLED(CONFIG_DRM_RADEON_MODULE)
-int amdgpu_si_support;
-MODULE_PARM_DESC(si_support, "SI support (1 = enabled, 0 = disabled (default))");
-#else
 int amdgpu_si_support = 1;
 MODULE_PARM_DESC(si_support, "SI support (1 = enabled (default), 0 = disabled)");
-#endif
 
 module_param_named(si_support, amdgpu_si_support, int, 0444);
 #endif
@@ -608,18 +602,12 @@ module_param_named(si_support, amdgpu_si
 /**
  * DOC: cik_support (int)
  * Set CIK support driver. This parameter works after set config CONFIG_DRM_AMDGPU_CIK. For CIK asic, when radeon driver is enabled,
- * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using radeon driver when it available,
- * otherwise using amdgpu driver.
+ * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using amdgpu driver.
  */
 #ifdef CONFIG_DRM_AMDGPU_CIK
 
-#if IS_ENABLED(CONFIG_DRM_RADEON) || IS_ENABLED(CONFIG_DRM_RADEON_MODULE)
-int amdgpu_cik_support;
-MODULE_PARM_DESC(cik_support, "CIK support (1 = enabled, 0 = disabled (default))");
-#else
 int amdgpu_cik_support = 1;
 MODULE_PARM_DESC(cik_support, "CIK support (1 = enabled (default), 0 = disabled)");
-#endif
 
 module_param_named(cik_support, amdgpu_cik_support, int, 0444);
 #endif
diff -up linux-6.9-rc1/drivers/gpu/drm/radeon/radeon_drv.c.28~ linux-6.9-rc1/drivers/gpu/drm/radeon/radeon_drv.c
--- linux-6.9-rc1/drivers/gpu/drm/radeon/radeon_drv.c.28~	2024-03-24 22:10:05.000000000 +0100
+++ linux-6.9-rc1/drivers/gpu/drm/radeon/radeon_drv.c	2024-03-25 12:27:48.258221857 +0100
@@ -239,11 +239,11 @@ module_param_named(uvd, radeon_uvd, int,
 MODULE_PARM_DESC(vce, "vce enable/disable vce support (1 = enable, 0 = disable)");
 module_param_named(vce, radeon_vce, int, 0444);
 
-int radeon_si_support = 1;
+int radeon_si_support = 0;
 MODULE_PARM_DESC(si_support, "SI support (1 = enabled (default), 0 = disabled)");
 module_param_named(si_support, radeon_si_support, int, 0444);
 
-int radeon_cik_support = 1;
+int radeon_cik_support = 0;
 MODULE_PARM_DESC(cik_support, "CIK support (1 = enabled (default), 0 = disabled)");
 module_param_named(cik_support, radeon_cik_support, int, 0444);
 
