diff -up linux-6.19-rc4/scripts/Makefile.9~ linux-6.19-rc4/scripts/Makefile
--- linux-6.19-rc4/scripts/Makefile.9~	2026-01-04 23:41:55.000000000 +0100
+++ linux-6.19-rc4/scripts/Makefile	2026-01-10 15:37:32.024037877 +0100
@@ -33,6 +33,7 @@ HOSTCFLAGS_tracepoint-update.o = -I$(src
 HOSTCFLAGS_elf-parse.o = -I$(srctree)/tools/include
 HOSTCFLAGS_sorttable.o = -I$(srctree)/tools/include
 HOSTLDLIBS_sorttable = -lpthread
+HOSTLDFLAGS_sorttable = --icf=none
 HOSTCFLAGS_asn1_compiler.o = -I$(srctree)/include
 HOSTCFLAGS_sign-file.o = $(shell $(HOSTPKG_CONFIG) --cflags libcrypto 2> /dev/null)
 HOSTLDLIBS_sign-file = $(shell $(HOSTPKG_CONFIG) --libs libcrypto 2> /dev/null || echo -lcrypto)
