diff -up linux-5.2-rc1/arch/arm64/kernel/vdso/Makefile.4~ linux-5.2-rc1/arch/arm64/kernel/vdso/Makefile
--- linux-5.2-rc1/arch/arm64/kernel/vdso/Makefile.4~	2019-05-21 02:50:12.484303922 +0200
+++ linux-5.2-rc1/arch/arm64/kernel/vdso/Makefile	2019-05-21 02:55:14.651708274 +0200
@@ -13,7 +13,7 @@ targets := $(obj-vdso) vdso.so vdso.so.d
 obj-vdso := $(addprefix $(obj)/, $(obj-vdso))
 
 ldflags-y := -shared -nostdlib -soname=linux-vdso.so.1 \
-		$(call ld-option, --hash-style=sysv) -n -T
+		$(call ld-option, --hash-style=sysv -fuse-ld=bfd) -n -T
 
 # Disable gcov profiling for VDSO code
 GCOV_PROFILE := n
diff -up linux-5.2-rc1/arch/x86/entry/vdso/Makefile.4~ linux-5.2-rc1/arch/x86/entry/vdso/Makefile
--- linux-5.2-rc1/arch/x86/entry/vdso/Makefile.4~	2019-05-20 00:47:09.000000000 +0200
+++ linux-5.2-rc1/arch/x86/entry/vdso/Makefile	2019-05-21 02:50:12.485303914 +0200
@@ -47,7 +47,7 @@ targets += $(vdso_img_sodbg) $(vdso_img-
 CPPFLAGS_vdso.lds += -P -C
 
 VDSO_LDFLAGS_vdso.lds = -m elf_x86_64 -soname linux-vdso.so.1 --no-undefined \
-			-z max-page-size=4096
+			-z max-page-size=4096 -fuse-ld=bfd
 
 $(obj)/vdso64.so.dbg: $(obj)/vdso.lds $(vobjs) FORCE
 	$(call if_changed,vdso)
@@ -98,7 +98,7 @@ CFLAGS_REMOVE_vvar.o = -pg
 
 CPPFLAGS_vdsox32.lds = $(CPPFLAGS_vdso.lds)
 VDSO_LDFLAGS_vdsox32.lds = -m elf32_x86_64 -soname linux-vdso.so.1 \
-			   -z max-page-size=4096
+			   -z max-page-size=4096 -fuse-ld=bfd
 
 # x32-rebranded versions
 vobjx32s-y := $(vobjs-y:.o=-x32.o)
