diff -up linux-5.19/scripts/Makefile.8~ linux-5.19/scripts/Makefile
--- linux-5.19/scripts/Makefile.8~	2022-08-01 21:19:18.131885545 +0200
+++ linux-5.19/scripts/Makefile	2022-08-01 21:22:58.109685458 +0200
@@ -13,6 +13,7 @@ hostprogs-always-$(CONFIG_SYSTEM_EXTRA_C
 
 HOSTCFLAGS_sorttable.o = -I$(srctree)/tools/include
 HOSTLDLIBS_sorttable = -lpthread
+HOSTLDFLAGS_sorttable = --icf=none
 HOSTCFLAGS_asn1_compiler.o = -I$(srctree)/include
 HOSTCFLAGS_sign-file.o = $(shell $(HOSTPKG_CONFIG) --cflags libcrypto 2> /dev/null)
 HOSTLDLIBS_sign-file = $(shell $(HOSTPKG_CONFIG) --libs libcrypto 2> /dev/null || echo -lcrypto)
