diff -up linux-7.0-rc1/lib/Kconfig.debug.4~ linux-7.0-rc1/lib/Kconfig.debug
--- linux-7.0-rc1/lib/Kconfig.debug.4~	2026-02-22 22:18:59.000000000 +0100
+++ linux-7.0-rc1/lib/Kconfig.debug	2026-02-23 04:38:47.136227200 +0100
@@ -398,7 +398,6 @@ config DEBUG_INFO_SPLIT
 config DEBUG_INFO_BTF
 	bool "Generate BTF type information"
 	depends on !DEBUG_INFO_SPLIT && !DEBUG_INFO_REDUCED
-	depends on !GCC_PLUGIN_RANDSTRUCT || COMPILE_TEST
 	depends on BPF_SYSCALL
 	depends on PAHOLE_VERSION >= 122
 	# pahole uses elfutils, which does not have support for Hexagon relocations
