diff -up linux-6.6-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c.27~ linux-6.6-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
--- linux-6.6-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c.27~	2023-09-11 13:54:46.832328983 +0200
+++ linux-6.6-rc1/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c	2023-09-11 13:56:44.440580941 +0200
@@ -577,18 +577,11 @@ module_param_named(timeout_period, amdgp
 /**
  * DOC: si_support (int)
  * Set SI support driver. This parameter works after set config CONFIG_DRM_AMDGPU_SI. For SI asic, when radeon driver is enabled,
- * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using radeon driver when it available,
- * otherwise using amdgpu driver.
+ * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using the amdgpu driver.
  */
 #ifdef CONFIG_DRM_AMDGPU_SI
-
-#if IS_ENABLED(CONFIG_DRM_RADEON) || IS_ENABLED(CONFIG_DRM_RADEON_MODULE)
-int amdgpu_si_support = 0;
-MODULE_PARM_DESC(si_support, "SI support (1 = enabled, 0 = disabled (default))");
-#else
 int amdgpu_si_support = 1;
 MODULE_PARM_DESC(si_support, "SI support (1 = enabled (default), 0 = disabled)");
-#endif
 
 module_param_named(si_support, amdgpu_si_support, int, 0444);
 #endif
@@ -596,18 +589,11 @@ module_param_named(si_support, amdgpu_si
 /**
  * DOC: cik_support (int)
  * Set CIK support driver. This parameter works after set config CONFIG_DRM_AMDGPU_CIK. For CIK asic, when radeon driver is enabled,
- * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using radeon driver when it available,
- * otherwise using amdgpu driver.
+ * set value 0 to use radeon driver, while set value 1 to use amdgpu driver. The default is using the amdgpu driver.
  */
 #ifdef CONFIG_DRM_AMDGPU_CIK
-
-#if IS_ENABLED(CONFIG_DRM_RADEON) || IS_ENABLED(CONFIG_DRM_RADEON_MODULE)
-int amdgpu_cik_support = 0;
-MODULE_PARM_DESC(cik_support, "CIK support (1 = enabled, 0 = disabled (default))");
-#else
 int amdgpu_cik_support = 1;
 MODULE_PARM_DESC(cik_support, "CIK support (1 = enabled (default), 0 = disabled)");
-#endif
 
 module_param_named(cik_support, amdgpu_cik_support, int, 0444);
 #endif
diff -up linux-6.6-rc1/drivers/gpu/drm/radeon/radeon_drv.c.27~ linux-6.6-rc1/drivers/gpu/drm/radeon/radeon_drv.c
--- linux-6.6-rc1/drivers/gpu/drm/radeon/radeon_drv.c.27~	2023-09-11 01:28:41.000000000 +0200
+++ linux-6.6-rc1/drivers/gpu/drm/radeon/radeon_drv.c	2023-09-11 13:54:46.832328983 +0200
@@ -239,11 +239,11 @@ module_param_named(uvd, radeon_uvd, int,
 MODULE_PARM_DESC(vce, "vce enable/disable vce support (1 = enable, 0 = disable)");
 module_param_named(vce, radeon_vce, int, 0444);
 
-int radeon_si_support = 1;
+int radeon_si_support = 0;
 MODULE_PARM_DESC(si_support, "SI support (1 = enabled (default), 0 = disabled)");
 module_param_named(si_support, radeon_si_support, int, 0444);
 
-int radeon_cik_support = 1;
+int radeon_cik_support = 0;
 MODULE_PARM_DESC(cik_support, "CIK support (1 = enabled (default), 0 = disabled)");
 module_param_named(cik_support, radeon_cik_support, int, 0444);
 
