diff -up linux-6.10-rc5/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_debugfs.c.omv~ linux-6.10-rc5/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_debugfs.c
--- linux-6.10-rc5/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_debugfs.c.omv~	2024-06-27 12:29:11.902223178 +0100
+++ linux-6.10-rc5/drivers/staging/vc04_services/interface/vchiq_arm/vchiq_debugfs.c	2024-06-27 12:29:41.172255520 +0100
@@ -138,7 +138,7 @@ void vchiq_debugfs_deinit(void)
 
 #else /* CONFIG_DEBUG_FS */
 
-void vchiq_debugfs_init(void)
+void vchiq_debugfs_init(struct vchiq_state *)
 {
 }
 
