diff -up linux-7.1-rc5/drivers/nexus/nexus_core.c.omv~ linux-7.1-rc5/drivers/nexus/nexus_core.c
--- linux-7.1-rc5/drivers/nexus/nexus_core.c.omv~	2026-05-27 14:52:45.572582431 +0200
+++ linux-7.1-rc5/drivers/nexus/nexus_core.c	2026-05-27 14:52:50.985168176 +0200
@@ -118,7 +118,7 @@ void nexus_unregister_team_exit(nexus_te
 }
 EXPORT_SYMBOL(nexus_unregister_team_exit);
 
-struct nexus_team* nexus_team_init()
+struct nexus_team* nexus_team_init(void)
 {
 	struct nexus_team* team = kzalloc(sizeof(struct nexus_team), GFP_KERNEL);
 	if (team != NULL) {
