diff -up linux-7.1-rc4/drivers/nexus/nexus_core.c.27~ linux-7.1-rc4/drivers/nexus/nexus_core.c
--- linux-7.1-rc4/drivers/nexus/nexus_core.c.27~	2026-05-12 02:16:32.000000000 +0200
+++ linux-7.1-rc4/drivers/nexus/nexus_core.c	2026-05-19 03:15:12.637468815 +0200
@@ -30,6 +30,9 @@ static int major = -1;
 static struct cdev nexus_cdev;
 static struct class *nexus_class = NULL;
 
+struct nexus_thread* find_thread(struct nexus_team *team, const char *name);
+struct nexus_thread* find_thread_by_id(struct nexus_team *team, int32_t pid);
+
 DEFINE_MUTEX(nexus_main_lock);
 HLIST_HEAD(nexus_teams);
 
