diff -up linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/os_intfs.c.omv~ linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/os_intfs.c
--- linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/os_intfs.c.omv~	2022-01-31 12:05:06.142083069 +0100
+++ linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/os_intfs.c	2022-01-31 12:17:48.836717810 +0100
@@ -1615,7 +1615,7 @@ static int rtw_net_set_mac_address(struc
 	}
 
 	_rtw_memcpy(adapter_mac_addr(padapter), sa->sa_data, ETH_ALEN); /* set mac addr to adapter */
-	_rtw_memcpy(pnetdev->dev_addr, sa->sa_data, ETH_ALEN); /* set mac addr to net_device */
+	_rtw_memcpy((void*)pnetdev->dev_addr, sa->sa_data, ETH_ALEN); /* set mac addr to net_device */
 
 #if 0
 	if (rtw_is_hw_init_completed(padapter)) {
@@ -2151,7 +2151,7 @@ int rtw_os_ndev_register(_adapter *adapt
 	/* alloc netdev name */
 	rtw_init_netdev_name(ndev, name);
 
-	_rtw_memcpy(ndev->dev_addr, adapter_mac_addr(adapter), ETH_ALEN);
+	eth_hw_addr_set(ndev, adapter_mac_addr(adapter));
 
 	/* Tell the network stack we exist */
 
@@ -3217,7 +3217,7 @@ int _netdev_vir_if_open(struct net_devic
 		rtw_mbid_camid_alloc(padapter, adapter_mac_addr(padapter));
 #endif
 		rtw_init_wifidirect_addrs(padapter, adapter_mac_addr(padapter), adapter_mac_addr(padapter));
-		_rtw_memcpy(pnetdev->dev_addr, adapter_mac_addr(padapter), ETH_ALEN);
+		eth_hw_addr_set(pnetdev, adapter_mac_addr(padapter));
 	}
 #endif /*CONFIG_PLATFORM_INTEL_BYT*/
 
diff -up linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/rtw_proc.c.omv~ linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/rtw_proc.c
--- linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/rtw_proc.c.omv~	2022-01-31 13:08:27.085659385 +0100
+++ linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/linux/rtw_proc.c	2022-01-31 13:08:45.621754054 +0100
@@ -35,7 +35,7 @@ inline struct proc_dir_entry *get_rtw_dr
 #endif
 
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 10, 0))
-#define PDE_DATA(inode) PDE((inode))->data
+#define pde_data(inode) PDE((inode))->data
 #define proc_get_parent_data(inode) PDE((inode))->parent->data
 #endif
 
@@ -247,7 +247,7 @@ const int drv_proc_hdls_num = sizeof(drv
 static int rtw_drv_proc_open(struct inode *inode, struct file *file)
 {
 	/* struct net_device *dev = proc_get_parent_data(inode); */
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = drv_proc_hdls + index;
 	void *private = NULL;
 
@@ -273,7 +273,7 @@ static int rtw_drv_proc_open(struct inod
 
 static ssize_t rtw_drv_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = drv_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
@@ -5631,7 +5631,7 @@ const int adapter_proc_hdls_num = sizeof
 
 static int rtw_adapter_proc_open(struct inode *inode, struct file *file)
 {
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = adapter_proc_hdls + index;
 	void *private = proc_get_parent_data(inode);
 
@@ -5657,7 +5657,7 @@ static int rtw_adapter_proc_open(struct
 
 static ssize_t rtw_adapter_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = adapter_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
@@ -5820,7 +5820,7 @@ const int odm_proc_hdls_num = sizeof(odm
 
 static int rtw_odm_proc_open(struct inode *inode, struct file *file)
 {
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = odm_proc_hdls + index;
 	void *private = proc_get_parent_data(inode);
 
@@ -5846,7 +5846,7 @@ static int rtw_odm_proc_open(struct inod
 
 static ssize_t rtw_odm_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = odm_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
@@ -5985,7 +5985,7 @@ const int mcc_proc_hdls_num = sizeof(mcc
 
 static int rtw_mcc_proc_open(struct inode *inode, struct file *file)
 {
-	ssize_t index = (ssize_t)PDE_DATA(inode);
+	ssize_t index = (ssize_t)pde_data(inode);
 	const struct rtw_proc_hdl *hdl = mcc_proc_hdls + index;
 	void *private = proc_get_parent_data(inode);
 
@@ -6011,7 +6011,7 @@ static int rtw_mcc_proc_open(struct inod
 
 static ssize_t rtw_mcc_proc_write(struct file *file, const char __user *buffer, size_t count, loff_t *pos)
 {
-	ssize_t index = (ssize_t)PDE_DATA(file_inode(file));
+	ssize_t index = (ssize_t)pde_data(file_inode(file));
 	const struct rtw_proc_hdl *hdl = mcc_proc_hdls + index;
 	ssize_t (*write)(struct file *, const char __user *, size_t, loff_t *, void *) = hdl->write;
 
diff -up linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/osdep_service.c.omv~ linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/osdep_service.c
--- linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/osdep_service.c.omv~	2022-01-31 12:20:54.250348135 +0100
+++ linux-5.17-rc2/drivers/staging/rtl8723cs/os_dep/osdep_service.c	2022-01-31 12:21:00.287368625 +0100
@@ -1309,7 +1309,7 @@ u32 _rtw_down_sema(_sema *sema)
 inline void thread_exit(_completion *comp)
 {
 #ifdef PLATFORM_LINUX
-	complete_and_exit(comp, 0);
+	kthread_complete_and_exit(comp, 0);
 #endif
 
 #ifdef PLATFORM_FREEBSD
