--- linux-7.2-rc6/drivers/android/binder.c	2026-08-03 01:24:24.000000000 +0200
+++ linux-7.2-rc6/drivers/android/binder.c	2026-08-09 23:35:42.277506921 +0200
@@ -7016,6 +7016,9 @@ DEFINE_SHOW_ATTRIBUTE(transactions);
 DEFINE_SHOW_ATTRIBUTE(transactions_hashed);
 DEFINE_SHOW_ATTRIBUTE(transaction_log);
 
+MODULE_DESCRIPTION("The binder IPC system");
+MODULE_LICENSE("GPL v2");
+
 const struct binder_debugfs_entry binder_debugfs_entries[] = {
 	{
 		.name = "state",

--- linux-7.2-rc6/mm/list_lru.c	2026-08-03 01:24:24.000000000 +0200
+++ linux-7.2-rc6/mm/list_lru.c	2026-08-09 23:35:42.278867157 +0200
@@ -273,6 +273,7 @@ bool list_lru_del(struct list_lru *lru,
 	list_lru_unlock(l);
 	return ret;
 }
+EXPORT_SYMBOL_GPL(list_lru_del); // Still used by binder
 
 bool list_lru_del_obj(struct list_lru *lru, struct list_head *item)
 {

--- linux-7.2-rc6/mm/mmap_lock.c	2026-08-03 01:24:24.000000000 +0200
+++ linux-7.2-rc6/mm/mmap_lock.c	2026-08-09 23:35:42.279253423 +0200
@@ -340,6 +340,7 @@ inval:
 	count_vm_vma_lock_event(VMA_LOCK_ABORT);
 	return NULL;
 }
+EXPORT_SYMBOL_GPL(lock_vma_under_rcu);
 
 static struct vm_area_struct *lock_next_vma_under_mmap_lock(struct mm_struct *mm,
 							    struct vma_iterator *vmi,

--- linux-7.2-rc6/mm/memory.c	2026-08-03 01:24:24.000000000 +0200
+++ linux-7.2-rc6/mm/memory.c	2026-08-09 23:35:42.280105387 +0200
@@ -2247,6 +2247,7 @@ void zap_vma_range(struct vm_area_struct
 	zap_vma_range_batched(&tlb, vma, address, size, NULL);
 	tlb_finish_mmu(&tlb);
 }
+EXPORT_SYMBOL_GPL(zap_vma_range);
 
 /**
  * zap_special_vma_range - zap all page table entries in a special vma range
