diff -up linux-6.17-rc1/drivers/gpu/drm/rockchip/dw-dp.c.omv~ linux-6.17-rc1/drivers/gpu/drm/rockchip/dw-dp.c
--- linux-6.17-rc1/drivers/gpu/drm/rockchip/dw-dp.c.omv~	2025-08-17 14:41:11.462124202 +0200
+++ linux-6.17-rc1/drivers/gpu/drm/rockchip/dw-dp.c	2025-08-17 14:42:18.980837608 +0200
@@ -574,10 +574,10 @@ dw_dp_connector_detect(struct drm_connec
 {
 	struct dw_dp *dp = connector_to_dp(connector);
 
-	if (dp->right && drm_bridge_detect(&dp->right->bridge) != connector_status_connected)
+	if (dp->right && drm_bridge_detect(&dp->right->bridge, connector) != connector_status_connected)
 		return connector_status_disconnected;
 
-	return drm_bridge_detect(&dp->bridge);
+	return drm_bridge_detect(&dp->bridge, connector);
 }
 
 static void dw_dp_connector_force(struct drm_connector *connector)
@@ -2458,7 +2458,7 @@ fail_power_on:
 	return false;
 }
 
-static enum drm_connector_status dw_dp_bridge_detect(struct drm_bridge *bridge)
+static enum drm_connector_status dw_dp_bridge_detect(struct drm_bridge *bridge, struct drm_connector *conn)
 {
 	struct dw_dp *dp = bridge_to_dp(bridge);
 	enum drm_connector_status status = connector_status_connected;
@@ -2480,7 +2480,7 @@ static enum drm_connector_status dw_dp_b
 		struct drm_bridge *next_bridge = dp->next_bridge;
 
 		if (next_bridge->ops & DRM_BRIDGE_OP_DETECT)
-			status = drm_bridge_detect(next_bridge);
+			status = drm_bridge_detect(next_bridge, conn);
 	}
 
 out:
