diff -up linux-6.12/drivers/gpu/drm/rockchip/dw-dp.c.omv~ linux-6.12/drivers/gpu/drm/rockchip/dw-dp.c
--- linux-6.12/drivers/gpu/drm/rockchip/dw-dp.c.omv~	2024-11-19 04:26:44.403720312 +0000
+++ linux-6.12/drivers/gpu/drm/rockchip/dw-dp.c	2024-11-19 04:26:51.393718946 +0000
@@ -8,7 +8,7 @@
  *         Andy Yan <andy.yan@rock-chips.com>
  */
 
-#include <asm/unaligned.h>
+#include <linux/unaligned.h>
 #include <drm/display/drm_dp_helper.h>
 #include <drm/display/drm_hdmi_helper.h>
 #include <drm/drm_atomic_helper.h>
@@ -3164,14 +3164,12 @@ static int dw_dp_probe(struct platform_d
 	return component_add(dev, &dw_dp_component_ops);
 }
 
-static int dw_dp_remove(struct platform_device *pdev)
+static void dw_dp_remove(struct platform_device *pdev)
 {
 	struct dw_dp *dp = platform_get_drvdata(pdev);
 
 	component_del(dp->dev, &dw_dp_component_ops);
 	cancel_work_sync(&dp->hpd_work);
-
-	return 0;
 }
 
 static int __maybe_unused dw_dp_runtime_suspend(struct device *dev)
