diff -up linux-7.0-rc3/net/core/secure_seq.c.28~ linux-7.0-rc3/net/core/secure_seq.c
--- linux-7.0-rc3/net/core/secure_seq.c.28~	2026-03-09 00:56:54.000000000 +0100
+++ linux-7.0-rc3/net/core/secure_seq.c	2026-03-10 16:48:48.988208563 +0100
@@ -51,26 +51,11 @@ union tcp_seq_and_ts_off
 secure_tcpv6_seq_and_ts_off(const struct net *net, const __be32 *saddr,
 			    const __be32 *daddr, __be16 sport, __be16 dport)
 {
-	const struct {
-		struct in6_addr saddr;
-		struct in6_addr daddr;
-		__be16 sport;
-		__be16 dport;
-	} __aligned(SIPHASH_ALIGNMENT) combined = {
-		.saddr = *(struct in6_addr *)saddr,
-		.daddr = *(struct in6_addr *)daddr,
-		.sport = sport,
-		.dport = dport
-	};
 	union tcp_seq_and_ts_off st;
 
-	net_secret_init();
-
-	st.hash64 = siphash(&combined, offsetofend(typeof(combined), dport),
-			    &net_secret);
-
 	if (READ_ONCE(net->ipv4.sysctl_tcp_timestamps) != 1)
 		st.ts_off = 0;
+	get_random_bytes(((char*)&st.hash64), sizeof(st.hash64));
 
 	st.seq = seq_scale(st.seq);
 	return st;
@@ -108,13 +93,9 @@ union tcp_seq_and_ts_off
 secure_tcp_seq_and_ts_off(const struct net *net, __be32 saddr, __be32 daddr,
 			  __be16 sport, __be16 dport)
 {
-	u32 ports = (__force u32)sport << 16 | (__force u32)dport;
 	union tcp_seq_and_ts_off st;
 
-	net_secret_init();
-
-	st.hash64 = siphash_3u32((__force u32)saddr, (__force u32)daddr,
-				 ports, &net_secret);
+	get_random_bytes(((char *)&st.hash64), sizeof(st.hash64));
 
 	if (READ_ONCE(net->ipv4.sysctl_tcp_timestamps) != 1)
 		st.ts_off = 0;
