diff -up linux-6.12-rc3/drivers/hid/tmff-new/src/hid-tmff2.c.omv~ linux-6.12-rc3/drivers/hid/tmff-new/src/hid-tmff2.c
--- linux-6.12-rc3/drivers/hid/tmff-new/src/hid-tmff2.c.omv~	2024-10-15 21:05:39.182460245 +0200
+++ linux-6.12-rc3/drivers/hid/tmff-new/src/hid-tmff2.c	2024-10-15 21:05:54.642612377 +0200
@@ -712,7 +712,7 @@ oom_err:
 	return ret;
 }
 
-static __u8 *tmff2_report_fixup(struct hid_device *hdev, __u8 *rdesc,
+static const __u8 *tmff2_report_fixup(struct hid_device *hdev, __u8 *rdesc,
 		unsigned int *rsize)
 {
 	struct tmff2_device_entry *tmff2 = tmff2_from_hdev(hdev);
