diff -up linux-5.16-rc3/drivers/nvme/host/core.c.12~ linux-5.16-rc3/drivers/nvme/host/core.c
--- linux-5.16-rc3/drivers/nvme/host/core.c.12~	2021-11-29 15:50:07.016683446 +0000
+++ linux-5.16-rc3/drivers/nvme/host/core.c	2021-11-29 17:20:59.844346875 +0000
@@ -2492,6 +2492,15 @@ static const struct nvme_core_quirk_entr
 		.vid = 0x1e0f,
 		.mn = "KCD6XVUL6T40",
 		.quirks = NVME_QUIRK_NO_APST,
+	},
+        {
+		/* https://forum.openmandriva.org/t/nvme-ssd-m2-not-seen-by-omlx-4-0/2407
+		 * Unknown SSD ... Maybe ADATA/Hynix (a similar mn from ADTA
+		 * but vid seems to be Hynix)
+		 */
+		.vid = 0x1c5c,
+		.mn = "HFM128GDHTNG-8310B",
+		.quirks = NVME_QUIRK_NO_APST,
 	}
 };
 
