diff -up linux-5.19/drivers/net/vboxnetadp/Makefile-header.gmk.1005~ linux-5.19/drivers/net/vboxnetadp/Makefile-header.gmk
--- linux-5.19/drivers/net/vboxnetadp/Makefile-header.gmk.1005~	2022-08-03 19:04:38.657271905 +0200
+++ linux-5.19/drivers/net/vboxnetadp/Makefile-header.gmk	2022-08-03 19:10:42.930549625 +0200
@@ -50,7 +50,7 @@
 #
 
 # VBOX_KBUILD_TARGET_ARCH = amd64|x86
-ifeq ($(filter-out x86_64 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
+ifeq ($(filter-out x86_64 znver1 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := amd64
 else
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := x86
diff -up linux-5.19/drivers/net/vboxnetflt/Makefile-header.gmk.1005~ linux-5.19/drivers/net/vboxnetflt/Makefile-header.gmk
--- linux-5.19/drivers/net/vboxnetflt/Makefile-header.gmk.1005~	2022-08-03 19:04:38.658271914 +0200
+++ linux-5.19/drivers/net/vboxnetflt/Makefile-header.gmk	2022-08-03 19:10:55.337660864 +0200
@@ -50,7 +50,7 @@
 #
 
 # VBOX_KBUILD_TARGET_ARCH = amd64|x86
-ifeq ($(filter-out x86_64 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
+ifeq ($(filter-out x86_64 znver1 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := amd64
 else
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := x86
diff -up linux-5.19/drivers/virt/vboxdrv/Makefile-header.gmk.1005~ linux-5.19/drivers/virt/vboxdrv/Makefile-header.gmk
--- linux-5.19/drivers/virt/vboxdrv/Makefile-header.gmk.1005~	2022-08-03 19:04:38.658271914 +0200
+++ linux-5.19/drivers/virt/vboxdrv/Makefile-header.gmk	2022-08-03 19:11:13.765826050 +0200
@@ -50,7 +50,7 @@
 #
 
 # VBOX_KBUILD_TARGET_ARCH = amd64|x86
-ifeq ($(filter-out x86_64 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
+ifeq ($(filter-out x86_64 znver1 amd64 AMD64,$(shell rpm --eval %_target_cpu)),)
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := amd64
 else
  VBOX_KBUILD_TARGET_ARCH_DEFAULT := x86
