diff -up linux-5.0-rc2/fs/vboxsf/super.c.omv~ linux-5.0-rc2/fs/vboxsf/super.c
--- linux-5.0-rc2/fs/vboxsf/super.c.omv~	2019-01-14 04:11:57.486326225 +0100
+++ linux-5.0-rc2/fs/vboxsf/super.c	2019-01-14 04:12:26.646828086 +0100
@@ -12,6 +12,7 @@
 
 #include <linux/magic.h>
 #include <linux/module.h>
+#include <uapi/linux/mount.h>
 #include <linux/nls.h>
 #include <linux/parser.h>
 #include <linux/statfs.h>
