diff -up linux-7.0/drivers/virt/vboxdrv/linux/SUPDrv-linux.c.omv~ linux-7.0/drivers/virt/vboxdrv/linux/SUPDrv-linux.c
--- linux-7.0/drivers/virt/vboxdrv/linux/SUPDrv-linux.c.omv~	2026-04-19 22:24:48.972985841 +0200
+++ linux-7.0/drivers/virt/vboxdrv/linux/SUPDrv-linux.c	2026-04-19 22:26:53.225276734 +0200
@@ -210,9 +210,9 @@ static bool
 /** Whether we need to put the corresponding KVM function pointers. */
 static bool                                     g_fPutKvmEnableVirtualization, g_fPutKvmDisableVirtualization;
 /** Function pointer to kvm_enable_virtualization(). */
-static __typeof__(kvm_enable_virtualization)   *g_pfnKvmEnableVirtualization;
+static __typeof__(kvm_arch_enable_virtualization)   *g_pfnKvmEnableVirtualization;
 /** Function pointer to kvm_disable_virtualization(). */
-static __typeof__(kvm_disable_virtualization)  *g_pfnKvmDisableVirtualization;
+static __typeof__(kvm_arch_disable_virtualization)  *g_pfnKvmDisableVirtualization;
 /** Pointer to the KVM hardware specific module. */
 static struct module                           *g_pKvmHwvirtModule;
 #endif
@@ -504,7 +504,7 @@ static int __init supdrvLinuxInitKvmSymb
                 RTThreadPreemptDisable(&Preempt);
                 if (   uLeaves >= 7
                     && RTX86IsValidStdRange(uLeaves)
-                    && (ASMCpuIdEx_EDX(7, 0) & X86_CPUID_STEXT_FEATURE_EDX_CET_IBT))
+                    && (VBoxHost_ASMCpuId_EDX(7) & X86_CPUID_STEXT_FEATURE_EDX_CET_IBT))
                 {
                     uint64_t const fSupCet = ASMRdMsr(MSR_IA32_S_CET);
                     ASMWrMsr(MSR_IA32_S_CET, fSupCet & ~MSR_IA32_CET_ENDBR_EN);
@@ -1959,15 +1959,10 @@ int VBOXCALL supdrvOSEnableHwvirt(bool f
     if (fEnable)
     {
         /* kvm_enable_virtualization() is guarded by kvm_usage_count reference counter inside a mutex. */
-        int const rc = g_pfnKvmEnableVirtualization();
-        if (!rc)
-        {
-            g_fEnabledHwvirtUsingKvm = true;
-            printk(KERN_INFO "vboxdrv: Enabled hardware-virtualization using KVM\n");
-            return VINF_SUCCESS;
-        }
-        printk(KERN_ERR "vboxdrv: Failed to enable hardware-virtualization using KVM. rc=%d\n", rc);
-        return VERR_NOT_AVAILABLE;
+        g_pfnKvmEnableVirtualization();
+        g_fEnabledHwvirtUsingKvm = true;
+        printk(KERN_INFO "vboxdrv: Enabled hardware-virtualization using KVM\n");
+        return VINF_SUCCESS;
     }
 
     if (g_fEnabledHwvirtUsingKvm)
