diff -up linux-5.10-rc1/drivers/pci/vboxpci/linux/VBoxPci-linux.c.omv~ linux-5.10-rc1/drivers/pci/vboxpci/linux/VBoxPci-linux.c
--- linux-5.10-rc1/drivers/pci/vboxpci/linux/VBoxPci-linux.c.omv~	2020-10-27 00:27:19.215002353 +0100
+++ linux-5.10-rc1/drivers/pci/vboxpci/linux/VBoxPci-linux.c	2020-10-27 00:29:11.139501605 +0100
@@ -369,16 +369,12 @@ static void  vboxPciFileClose(struct fil
 static int vboxPciFileWrite(struct file* file, unsigned long long offset, unsigned char* data, unsigned int size)
 {
     int          ret;
-    mm_segment_t fs_save;
 
-    fs_save = get_fs();
-    set_fs(KERNEL_DS);
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
     ret = kernel_write(file, data, size, &offset);
 #else
     ret = vfs_write(file, data, size, &offset);
 #endif
-    set_fs(fs_save);
     if (ret < 0)
         printk(KERN_DEBUG "vboxPciFileWrite: error %d\n", ret);
 
