diff -up linux-6.5-rc1/drivers/net/vboxnetflt/linux/VBoxNetFlt-linux.c.omv~ linux-6.5-rc1/drivers/net/vboxnetflt/linux/VBoxNetFlt-linux.c
--- linux-6.5-rc1/drivers/net/vboxnetflt/linux/VBoxNetFlt-linux.c.omv~	2023-07-15 00:50:16.315198405 +0200
+++ linux-6.5-rc1/drivers/net/vboxnetflt/linux/VBoxNetFlt-linux.c	2023-07-15 00:50:29.633340272 +0200
@@ -64,6 +64,7 @@
 #include <net/ipv6.h>
 #include <net/if_inet6.h>
 #include <net/addrconf.h>
+#include <net/gso.h>
 
 #include <VBox/log.h>
 #include <VBox/err.h>
diff -up linux-6.5-rc1/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c.omv~ linux-6.5-rc1/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c
--- linux-6.5-rc1/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c.omv~	2023-07-15 01:10:40.462742870 +0200
+++ linux-6.5-rc1/drivers/virt/vboxdrv/r0drv/linux/memobj-r0drv-linux.c	2023-07-15 01:20:48.388527301 +0200
@@ -1335,8 +1335,7 @@ DECLHIDDEN(int) rtR0MemObjNativeLockUser
                                 fWrite,                 /* Write to memory. */
                                 fWrite,                 /* force write access. */
 # endif
-                                &pMemLnx->apPages[0],   /* Page array. */
-                                papVMAs);               /* vmas */
+                                &pMemLnx->apPages[0]);  /* Page array. */
         /*
          * Actually this should not happen at the moment as call this function
          * only for our own process.
@@ -1358,9 +1357,8 @@ DECLHIDDEN(int) rtR0MemObjNativeLockUser
                                 fWrite,                 /* force write access. */
 # endif
                                 &pMemLnx->apPages[0],   /* Page array. */
-                                papVMAs                 /* vmas */
 # if GET_USER_PAGES_API >= KERNEL_VERSION(4, 10, 0)
-                                , NULL                  /* locked */
+                                NULL                    /* locked */
 # endif
                                 );
 #else /* GET_USER_PAGES_API < KERNEL_VERSION(4, 6, 0) */
diff -up linux-6.5-rc1/include/linux/mm.h.omv~ linux-6.5-rc1/include/linux/mm.h
diff -up linux-6.5-rc1/mm/pgtable-generic.c.omv~ linux-6.5-rc1/mm/pgtable-generic.c
--- linux-6.5-rc1/mm/pgtable-generic.c.omv~	2023-07-15 01:58:41.991312027 +0200
+++ linux-6.5-rc1/mm/pgtable-generic.c	2023-07-15 01:58:56.546483707 +0200
@@ -253,6 +253,7 @@ nomap:
 	/* rcu_read_unlock() to be added later */
 	return NULL;
 }
+EXPORT_SYMBOL_GPL(__pte_offset_map);
 
 pte_t *pte_offset_map_nolock(struct mm_struct *mm, pmd_t *pmd,
 			     unsigned long addr, spinlock_t **ptlp)
